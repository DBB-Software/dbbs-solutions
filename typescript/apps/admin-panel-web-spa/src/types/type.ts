export type Type = {
  id: string
  name: string
  createdAt: string
  updatedAt: string
}
