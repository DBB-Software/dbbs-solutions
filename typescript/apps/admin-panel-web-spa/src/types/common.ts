import { GridSortDirection } from '@mui/x-data-grid'

export type UpdatePayload<T extends object> = {
  id: string
  item: Partial<T>
}

export type ListPayload = {
  limit?: number
  offset?: number
  filterField?: string
  filterOperator?: string
  filterValue?: string
  sortBy?: string
  sortOrder?: GridSortDirection
}

export type ListResponse<T extends object> = {
  cursor: number
  count: number
  remaining: number
  results: T[]
}

export type LinkField = {
  id: string
  name: string
  fullName: string
}
