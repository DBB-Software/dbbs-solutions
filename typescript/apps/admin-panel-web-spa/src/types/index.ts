export * from './common'
export * from './geo'
export * from './meetingType'
