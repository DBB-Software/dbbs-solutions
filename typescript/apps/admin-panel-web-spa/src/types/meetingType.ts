export type MeetingType = {
  id: string
  name: string
  createdAt: string
  updatedAt: string
}
