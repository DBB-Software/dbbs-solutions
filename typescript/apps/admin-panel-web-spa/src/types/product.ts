import { Type } from './type'

export type Product = {
  id: number
  typeId?: number
  name?: string
  description?: string
  price?: number
  currency?: string
  sku?: string
  inventoryCount?: number
  imageUrl?: string
  isActive?: boolean
  createdAt?: string
  updatedAt?: string
  type?: Type
}
