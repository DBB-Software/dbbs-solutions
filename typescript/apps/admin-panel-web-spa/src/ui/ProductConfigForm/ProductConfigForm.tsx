import React, { FormEventHandler, FC } from 'react'
import { Controller, Control, FieldErrors, FieldValues } from 'react-hook-form'
import { TextField, Typography, FormControl, FormControlLabel, Checkbox, Button, Grid } from '@dbbs/mui-components'
import { CustomAutocomplete, Option } from '../CustomAutocomplete'
import { DEFAULT_CONFIG_PAGE_GRID_SIZE } from '../../utils'
import { GEO_CONFIG_FORM_TEST_IDS } from './testIds'

export interface ProductConfigFormProps {
  isUpdateLoading: boolean
  control?: Control<FieldValues>
  handleSubmit: FormEventHandler<HTMLFormElement>
  errors: FieldErrors
  onTypeSearch: (searchTerm: string) => void
  typeSearchLoading: boolean
  typeOptions: Option[]
}

export const ProductConfigForm: FC<ProductConfigFormProps> = ({
  isUpdateLoading,
  control,
  handleSubmit,
  errors,
  onTypeSearch,
  typeSearchLoading,
  typeOptions
}) => (
  <form onSubmit={handleSubmit} data-testid={GEO_CONFIG_FORM_TEST_IDS.FORM}>
    <Grid container spacing={2}>
      <Grid xs={12} item>
        <Typography variant="h4">Product Configuration</Typography>
      </Grid>

      <Grid {...DEFAULT_CONFIG_PAGE_GRID_SIZE} item>
        <Controller
          name="name"
          control={control}
          rules={{ required: 'Name is required' }}
          render={({ field }) => (
            <FormControl fullWidth>
              <TextField
                {...field}
                label="Name"
                error={!!errors.name}
                fullWidth
                data-testid={GEO_CONFIG_FORM_TEST_IDS.NAME}
                InputLabelProps={{ shrink: !!field.value }}
              />
            </FormControl>
          )}
        />
      </Grid>

      <Grid {...DEFAULT_CONFIG_PAGE_GRID_SIZE} item>
        <Controller
          name="description"
          control={control}
          render={({ field }) => (
            <FormControl fullWidth>
              <TextField {...field} label="Description" fullWidth />
            </FormControl>
          )}
        />
      </Grid>

      <Grid {...DEFAULT_CONFIG_PAGE_GRID_SIZE} item>
        <Controller
          name="price"
          control={control}
          render={({ field }) => (
            <FormControl fullWidth>
              <TextField {...field} label="Price" type="number" fullWidth InputLabelProps={{ shrink: !!field.value }} />
            </FormControl>
          )}
        />
      </Grid>

      <Grid {...DEFAULT_CONFIG_PAGE_GRID_SIZE} item>
        <Controller
          name="currency"
          control={control}
          render={({ field }) => (
            <FormControl fullWidth>
              <TextField {...field} label="Currency" fullWidth InputLabelProps={{ shrink: !!field.value }} />
            </FormControl>
          )}
        />
      </Grid>

      <Grid {...DEFAULT_CONFIG_PAGE_GRID_SIZE} item>
        <Controller
          name="sku"
          control={control}
          render={({ field }) => (
            <FormControl fullWidth>
              <TextField {...field} label="SKU" fullWidth InputLabelProps={{ shrink: !!field.value }} />
            </FormControl>
          )}
        />
      </Grid>

      <Grid {...DEFAULT_CONFIG_PAGE_GRID_SIZE} item>
        <Controller
          name="inventoryCount"
          control={control}
          render={({ field }) => (
            <FormControl fullWidth>
              <TextField
                {...field}
                label="Inventory Count"
                type="number"
                fullWidth
                InputLabelProps={{ shrink: !!field.value }}
              />
            </FormControl>
          )}
        />
      </Grid>

      <Grid {...DEFAULT_CONFIG_PAGE_GRID_SIZE} item>
        <Controller
          name="imageUrl"
          control={control}
          render={({ field }) => (
            <FormControl fullWidth>
              <TextField {...field} label="Image URL" fullWidth InputLabelProps={{ shrink: !!field.value }} />
            </FormControl>
          )}
        />
      </Grid>

      <Grid {...DEFAULT_CONFIG_PAGE_GRID_SIZE} item>
        <FormControlLabel
          control={
            <Controller
              name="isActive"
              control={control}
              render={({ field }) => <Checkbox {...field} checked={!!field.value} />}
            />
          }
          label="Is Active"
        />
      </Grid>

      <Grid {...DEFAULT_CONFIG_PAGE_GRID_SIZE} item>
        <CustomAutocomplete
          name="type"
          control={control}
          label="Type"
          options={typeOptions}
          onSearch={onTypeSearch}
          loading={typeSearchLoading}
          idKey="id"
          labelKey="name"
        />
      </Grid>

      <Grid xs={12} item>
        <Button type="submit" variant="contained" fullWidth disabled={isUpdateLoading}>
          {isUpdateLoading ? 'Loading...' : 'Save'}
        </Button>
      </Grid>
    </Grid>
  </form>
)
