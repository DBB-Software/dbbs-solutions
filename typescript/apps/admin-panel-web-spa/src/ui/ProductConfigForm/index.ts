export * from './ProductConfigForm'
export * from './testIds'
