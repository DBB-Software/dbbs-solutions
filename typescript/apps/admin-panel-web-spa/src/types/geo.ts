import { LinkField } from './common'

export type Geo = {
  id: string
  bubbleId: string
  modifiedDate?: string
  createdDate?: string
  createdBy?: string
  name: string
  scheduleUrl: string
  geoType: string
  timezone?: string
  captureScheduleFlag?: boolean
  captureStreamFlag?: boolean
  scheduleFormat?: string
  streamType?: string
  jurisdiction?: boolean
  parent?: LinkField
  geoParentId?: number
  channelUrl?: string
  statusSchedule?: string
  detectStartMethod: string
  detectEndMethod: string
  statusStream?: string
  flagOnlyAgenda?: boolean
  flagOptInOnly?: boolean
  singlePlayerUrl?: string
  flagLive?: boolean
  detectEndOcrString?: string
  debug?: boolean
  demo?: boolean
  scheduleRefreshFrequency?: number
  streamTypeId?: number
}
export type GeoProperties = {
  geoTypes: string[]
  detectMethods: string[]
  statuses: string[]
  timezones: string[]
  scheduleFormats: string[]
  streamTypes: string[]
}
