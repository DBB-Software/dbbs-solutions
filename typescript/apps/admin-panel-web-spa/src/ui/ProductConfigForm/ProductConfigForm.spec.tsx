import React from 'react'
import { render } from '@testing-library/react'
import { FieldValues, useForm } from 'react-hook-form'
import { ProductConfigForm, ProductConfigFormProps } from './ProductConfigForm'
import { Product } from '../../types'

const FormWithControl = () => {
  const { control } = useForm<FieldValues, Product>({
    defaultValues: {}
  })
  const defaultProps: ProductConfigFormProps = {
    isUpdateLoading: false,
    handleSubmit: jest.fn(),
    errors: {},
    control,
    onTypeSearch: jest.fn(),
    typeSearchLoading: false,
    typeOptions: []
  }

  return <ProductConfigForm {...defaultProps} />
}

describe('<ProductConfigForm />', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<FormWithControl />)
    expect(asFragment()).toMatchSnapshot()
  })
})
