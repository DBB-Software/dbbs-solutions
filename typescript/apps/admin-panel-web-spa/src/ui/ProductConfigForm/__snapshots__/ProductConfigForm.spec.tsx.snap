// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductConfigForm /> renders correctly 1`] = `
<DocumentFragment>
  <form
    data-testid="product-config-form"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
        >
          Product Configuration
        </h4>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            data-testid="product-config-name"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r0:"
              id=":r0:-label"
            >
              Name
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r0:"
                name="name"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Name
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r1:"
              id=":r1:-label"
            >
              Description
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r1:"
                name="description"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Description
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r2:"
              id=":r2:-label"
            >
              Price
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r2:"
                name="price"
                type="number"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Price
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r3:"
              id=":r3:-label"
            >
              Currency
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r3:"
                name="currency"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Currency
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r4:"
              id=":r4:-label"
            >
              SKU
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r4:"
                name="sku"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    SKU
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r5:"
              id=":r5:-label"
            >
              Inventory Count
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r5:"
                name="inventoryCount"
                type="number"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Inventory Count
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r6:"
              id=":r6:-label"
            >
              Image URL
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r6:"
                name="imageUrl"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Image URL
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
      >
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              name="isActive"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxOutlineBlankIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
          >
            Is Active
          </span>
        </label>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
      >
        <div
          class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-ka7ti6-MuiAutocomplete-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r7:"
              id=":r7:-label"
            >
              Type
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r7:"
                role="combobox"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="MuiAutocomplete-endAdornment css-p1olib-MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Open"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                  tabindex="-1"
                  title="Open"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Type
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Save
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
